### PR TITLE
MM-37128 Detox/E2E Maintenance: Fix failing tests - iOS

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- These are maintenance fixes. Ran the suite and fixed failing tests found locally.
- Moved `detox/e2e/support/utils.js` to `detox/e2e/support/utils/index.js`
- Added email utils
    - `detox/e2e/support/utils/email.js`

Note: Will run the suite against android and will create a separate PR for that.

#### Ticket Link
JIRA tickets:
- https://mattermost.atlassian.net/browse/MM-37128

#### Screenshots
Screen Shot 2021-07-12 at 3 35 14 PM
Screen Shot 2021-07-12 at 4 10 13 PM
Screen Shot 2021-07-12 at 12 46 56 PM
Screen Shot 2021-07-15 at 2 15 42 PM
Screen Shot 2021-07-15 at 2 47 18 PM
Screen Shot 2021-07-15 at 2 55 41 PM
Screen Shot 2021-07-15 at 3 07 39 PM
Screen Shot 2021-07-15 at 3 25 44 PM
Screen Shot 2021-07-15 at 3 39 03 PM
Screen Shot 2021-07-15 at 4 23 33 PM
Screen Shot 2021-07-15 at 4 29 08 PM
Screen Shot 2021-07-15 at 4 38 37 PM
Screen Shot 2021-07-15 at 4 53 09 PM
Screen Shot 2021-07-15 at 4 56 06 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: detox/e2e/support/utils/email.js [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+};
+
+/**
+ * Split email body text.
+ * @param {string} text
+ * @return {string} split text
+ */
+export const splitEmailBodyText = (text) => {
+    return text.split('\n').map((d) => d.trim());
+};
```
Comment (saturninoabril) [inline detox/e2e/support/utils/email.js:238]: :100:

File: detox/e2e/test/main_sidebar/main_sidebar.e2e.js [hunk 2]
```diff
@@ -70,4 +56,18 @@ describe('Main Sidebar', () => {
...
+
+        // # Go to the same channel again
+        await goToChannel(testChannel.display_name);
+
+        // * Verify sidebar menu is not open
+        await expect(MainSidebar.mainSidebar).not.toBeVisible();
+
+        // * Selected channel should remain the same
+        await expect(channelNavBarTitle).toHaveText(testChannel.display_name);
+    });
```
Comment (saturninoabril) [inline detox/e2e/test/main_sidebar/main_sidebar.e2e.js:72]: Why move it here?

**Review by saturninoabril — APPROVED** [review]
Awesome, thanks @josephbaylon! Just one question but all looks good!

**Comment by enahum** [comment]
What are these email templates? And how does it affect the mobile app?

**Comment by saturninoabril** [comment]
Those are use to verify correct email notifications. We have it on Webapp/Cypress as well.

**Comment by enahum** [comment]
And how does it affect the mobile app? Mobile has nothing to do with emails, so I need more information about this

**Comment by saturninoabril** [comment]
See [MM-T3256](https://automation-test-cases.vercel.app/test/MM-T3256) for one example of test criteria where email notifications in Mobile app is being set to immediately and never. When set to immediately, we need to have a way to verify that the actual email is received and has expected content (since there could be other email message too). If set to never, then no email notification should have been sent. Without those email templates, the verification would not be complete if we'll only rely on mobile UI indication.

**Comment by enahum** [comment]
@saturninoabril that is exactly the context I needed!! Love it, thank you

**Review by enahum — APPROVED** [review]

**Pull request merged**